source/lac/trilinos_tpetra_vector.cc

```diff
@@ -0,0 +1,683 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 - 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/std_cxx14/memory.h>
+
+#include <deal.II/lac/trilinos_tpetra_vector.h>
+
+#ifdef DEAL_II_TRILINOS_WITH_TPETRA
+
+#  ifdef DEAL_II_WITH_MPI
+
+#    include <deal.II/base/index_set.h>
+
+#    include <deal.II/lac/read_write_vector.h>
+
+#    include <boost/io/ios_state.hpp>
+
+#    include <Teuchos_DefaultMpiComm.hpp>
+#    include <Tpetra_Import_def.hpp>
+#    include <Tpetra_Map_def.hpp>
+
+#    include <memory>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace LinearAlgebra
+{
+  namespace TpetraWrappers
+  {
+    template <typename Number>
```

Comment:
Would it make sense to move the implementation to `trilinos_tpetra_vector.templates.h` in case users want to use a type we didn't do any ETI for? Tpetra supports (with some work of the user) more type than what we instantiate.

Reply:
Sure, I'll do that.